src/OpenTelemetry/Trace/Builder/TracerProviderBuilderState.cs

```diff
@@ -109,11 +109,20 @@ public void SetResourceBuilder(ResourceBuilder resourceBuilder)
             this.ResourceBuilder = resourceBuilder;
         }
 
-        public void SetSampler(Sampler sampler)
+        public void ConfigureSampler(Action<SamplerBuilder> configure)
         {
-            Debug.Assert(sampler != null, "sampler was null");
+            Debug.Assert(configure != null, "configure was null");
+
+            var samplerBuilder = this.SamplerBuilder ??= SamplerBuilder.CreateDefault();
+
+            configure!(samplerBuilder);
+        }
+
+        public void SetSamplerBuilder(SamplerBuilder samplerBuilder)
+        {
+            Debug.Assert(samplerBuilder != null, "samplerBuilder was null");
```

Comment:
Shouldn't we use `Guard`?

```suggestion
            Guard.ThrowIfNull(samplerBuilder);
```